functions/Test-DbaLastBackup.ps1

```diff
@@ -259,7 +268,37 @@ Restores data and log files to alternative locations and only restores databases
 					}
 					
 					$lastbackup = Get-DbaBackupHistory -SqlServer $sourceserver -Databases $dbname -LastFull -IgnoreCopyOnly:$ignorecopyonly
-										
+
+					if($lastbackup[0].Path.StartsWith('\\') -eq $false -and $CopyDestination -eq $true) 
+					{
+						Write-Message -Level Verbose -Message "Copying backup to destination server"
+						if($destserver.BackupDirectory.StartsWith('\\') -eq $false)
+						{
+							Write-Message -Level Verbose -Message "Destination backup directory is not UNC and source does not match destination."
+							$fileexists = "Skipped"
+							$restoreresult = "Destination backup not on shared location."
+							$dbccresult = "Skipped"
+						}
+						elseif ((Test-SqlPath -SqlServer $destserver -path $destserver.BackupDirectory) -eq $true) 
+						{
+							if((Test-SqlPath -SqlServer $destserver -path ("{0}\{1}" -f $destserver.BackupDirectory,$Prefix)) -eq $false)
+							{
+								New-Item -Path ("{0}\{1}" -f $destserver.BackupDirectory,$Prefix) -ItemType Directory | Out-Null
+							}
+							Copy-Item -Path ("\\{0}\{1}" -f $lastbackup.Computername, $lastbackup.Path.replace(':','$')) -Destination ("{0}\{1}\{2}" -f $destserver.BackupDirectory,$Prefix,$lastbackup.path.split('\')[-1])
```

Comment:
Somewhat:
```Powerhell
try
{
    Copy-Item -Path ("\\{0}\{1}" -f $lastbackup.Computername, $lastbackup.Path.replace(':', '$')) -Destination "<Whatever desination was supposed to go here>" -ErrorAction Stop
}
catch
{
    Stop-Function -Message "Failed to copy backups to : " -Continue
}
```
What you will do in the catch really depends, on what should be done. If it makes no sense to continue with the current job, then yes, use Stop-Function (Its `-Continue` switch makes the function call `continue`, which stops the current job and starts the next Backup-Test).
Also - what you can't really already know, sorry for the complexity - Stop-Function supports some deeper debugging help, so the full line should be something like this:
`Stop-Function -Message "Failed to copy backups to : " -Target $instance -InnerErrorRecord $_ -Continue`
This will attach the instance object to the error log, as well as the actual exception thrown by Copy-Item.

If you want to simply continue the processing after something failed - for example the cleanup later in the script - instead of Stop-Function you use Write-Message for that:

```PowerShell
try
{
    Copy-Item -Path ("\\{0}\{1}" -f $lastbackup.Computername, $lastbackup.Path.replace(':', '$')) -Destination "<Whatever desination was supposed to go here>" -ErrorAction Stop
}
catch
{
    Write-Message -Level Warning -Message "Failed to copy backups to : <Destinatiob>" -ErrorRecord $_ -Target $instance
}
```
This will write a warning - or a non-terminating exception if in silent mode - and let the script continue.